### PR TITLE
fix(codegen,cli): generated output that compiles, refinements that don't abort the run, and a --no-codegen that takes effect

### PR DESCRIPTION
**Pull request by prisis**
Four fixes, one theme: the generator emits code that does not compile, or refuses to run at all, and the flag meant to stop it from overwriting your repairs does nothing. Two came from filed tickets; two were found while checking whether more of the same class was hiding.

Each commit stands alone and can be split out if you would rather review them separately.

## 1. `_generated/**` carried a pnpm store path (#497)

The type checker prints a type it resolved inside `node_modules` as the path it found on disk. Under pnpm that is a store path with a content hash in it, so any handler returning the ORM facade's `LoadWith<…>` — every global-table read — emitted:

```ts
import(".pnpm/@lunora+server@1.0.0-alpha.87_…/node_modules/@lunora/server/data-model")
```

Not a specifier, resolves from nowhere, and the hash changes on any `node_modules` rebuild — a TS2307 in generated output on the machine that produced it, and a different one everywhere else.

Everything after the final `node_modules/` is the specifier that was wanted, whatever the store nested in front of it. Recovering it also lets the existing umbrella rewrite finish the job: `@lunora/server/data-model` → `lunorash/server/data-model`, the name a consumer can actually resolve under pnpm's strict layout (`@lunora/server` is not reachable from a project that depends on `lunorash`, so the corrected-but-unmapped specifier would still fail).

Placed first in `rebaseRelativeQualifiers`, the one funnel every rendered args/return/chunk type passes through — ahead of the relative rebasers, which would otherwise read a `../../node_modules/…` form as one of the user's own modules and rebase plus `.js`-suffix it into something further still from a specifier.

**Why no example caught it:** in this monorepo every package is a workspace link, so the checker prints the bare package name and the umbrella rewrite handles it. The store path only appears where packages are installed from the registry. Verified by regenerating `examples/team-chat` with and without the fix.

## 2. `lunora dev --no-codegen` was inert (#496)

The flag did not stop anything: `_generated/**` was rewritten on every source save, with nothing logged, so the first sign was unrelated work failing to compile minutes later. Anything post-processing generated output — including a repair script for the bug above — was destroyed silently.

Root cause: on the Vite flavors the codegen watch lives in `@lunora/vite`, inside the framework dev server the CLI spawns as a **separate process**. `codegenEnabled: false` only ever meant "the CLI runs no watch of its own"; the flag never reached the process that owns the watch. Three paths dropped it:

| Path | Fix |
| --- | --- |
| `planDevCommand` (vite / framework-worker) | forwards `LUNORA_CODEGEN=0` to the child, next to the existing `LUNORA_REMOTE` hop |
| `startBackground`'s vite branch | spawns the dev script directly, bypassing `planDevCommand` — forwards the env itself. This is the path an AI-agent or `--background` start takes, which made it the default way to lose the flag |
| `ensureSidecarGenerated` | no longer runs its one-shot generate when the flag is set |

`@lunora/vite` honours it in `runCodegenSafely` — the single funnel both the startup run and the watch run pass through — and **logs the skip**. Silence was half the bug. Binding provisioning and the config-drift restart are untouched; only codegen output is gated.

`LUNORA_CODEGEN` is public and settable by hand, for a project that runs its framework dev script directly rather than through `lunora dev`.

## 3. `enum` and `class` return types emitted undeclared identifiers

Found by probing, not reported. The rule deciding whether the checker prints a user-land type as a bare name recognised only `interface` and `type` declarations, so a handler returning an `enum` or a `class` emitted:

```ts
status:  FunctionReference<"query", {}, { status: Status; }>;
receipt: FunctionReference<"query", {}, Receipt | undefined>;
```

TS2304 on both, in generated code, while `lunora codegen` exits 0. Reproduced by adding one such handler to `examples/todo-app` and running `tsc --noEmit`.

Both kinds now expand structurally like every other unreachable local type — a class to its instance shape (what a caller actually receives once serialized), an enum to its members' **values** (`"done" | "open"`, `1 | 2`), since the enum type itself prints as `Status.Done`, still bare. An enum reaches the rule as its *member* declarations, and a member is imported under its enum's name, never its own — so the reachability check asks about the parent.

## 4. `v.string().max(200)` aborted the whole run

```
error Unsupported validator kind: max
```

`.min` / `.max` / `.length` / `.pattern` / `.email` / `.url` / `.int` / `.positive` are published on `StringColumnValidator`, `NumberColumnValidator` and `ArrayColumnValidator`, and `@lunora/seed` already reads their constraints — but the parser knew only `.check(...)` and `.meta(...)` as transparent, so every other refinement hit the builder-member parser as if `.max` were a validator factory and threw. On a **schema column** as readily as on an `.input()` arg, so one documented call makes the app ungeneratable: no dev, no build, no deploy.

Each is a `self.check(...)` at runtime, so each unwraps to the receiver's kind and records `hasRefinement` — the AOT validator compiler must decline a node whose predicate the IR cannot carry, or the fast path would accept a 10,000-character string the interpreted parser rejects.

**And the shorthand spelling.** `.input({ bounded })` degraded the argument to `unknown` in the public api surface while `.input({ bounded: bounded })` resolved correctly. A shorthand property is its own initializer, so the identifier reaching alias resolution is the property *name*, whose symbol is the property rather than the const it stands for. Silent, and the wrong way round — `object-shorthand` autofixes the working spelling into the broken one, so a lint run could erase an argument's type. A schema test asserted the degraded `any` for exactly this shape; it now asserts the real kind.

## Not included

The post-codegen hook floated under #496's "Related". `postcodegen` already exists for `prepare` / `deploy` but is not wired into the dev watch. With #497 fixed the repair script it was meant to protect should no longer be needed — happy to add it if the need survives.

## Verification

- `codegen` / `cli` / `vite` / `config` / `seed`: all green, with a new test per fix.
- Committed `_generated/**` for **all 13 examples** is byte-identical after each change, regenerated **from cold** (`rm -rf lunora/_generated`) — which also re-confirms #283 stayed fixed.
- `lint:types`, `lint:eslint`, `api:check` clean; `config.api.md` updated for the two new exports.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

https://claude.ai/code/session_01Q4uV6jUTT13Hn1NdF5rWa1

## Summary by CodeRabbit

- **New Features**
  - Added support for disabling code generation with `--no-codegen` or `LUNORA_CODEGEN=0`.
  - The setting applies consistently across Vite development processes and file watching.
  - Added support for server-side default validators, making fields optional on insert.

- **Bug Fixes**
  - Prevented disabled code generation from modifying generated files.
  - Improved generated types for nested dependencies, enums, and value refinements.
  - Fixed path handling and shorthand validator definitions that could produce incomplete types.

- **Documentation**
  - Documented how to disable code generation when running development scripts directly.

### CODE REVIEW

File: packages/cli/src/commands/dev/index.ts [hunk 1]
```diff
@@ -47,7 +47,7 @@ const devCommand: Command = {
             name: "no-worker",
             type: Boolean,
         },
-        { description: "Don't watch + regenerate codegen", name: "no-codegen", type: Boolean },
+        { description: "Don't watch + regenerate codegen (or set LUNORA_CODEGEN=0)", name: "no-codegen", type: Boolean },
```
[the remark this answers is not in the document]
Comment (prisis) [reply packages/cli/src/commands/dev/index.ts]: Took the second option you offered — the CLI watcher now reads the contract rather than the help text being narrowed.

`codegenRequested` in `lifecycle.ts` is `options.codegen !== false && !isCodegenDisabled(process.env[CODEGEN_ENV])`, and both `runsCodegenWatch` (the plan field formerly called `codegenEnabled`) and `ensureSidecarGenerated` go through it. So `LUNORA_CODEGEN=0 lunora dev` now disables codegen on the wrangler flavor too, which is what the description and the docs already claimed. Covered by a test asserting `runsCodegenWatch` is false with the env set and true without it.

File: packages/codegen/src/emit.ts [hunk 2]
```diff
@@ -568,6 +568,41 @@
...
+ * A `@types/*` tail is never a specifier: the declarations for `foo` live in
+ * `@types/foo`, but the name anything imports is `foo`. Recovering the directory
+ * name would emit a package that does not exist, replacing a path that at least
+ * resolved locally — so leave the qualifier alone and let the relative rebasers
+ * have it.
+ */
+const TYPES_PACKAGE_SPEC_RE = /^@types\//u;
+
+const unresolveStoreQualifiers = (rendered: string): string =>
+    rendered.replaceAll(STORE_PATH_QUALIFIER_RE, (match, spec: string) => (TYPES_PACKAGE_SPEC_RE.test(spec) ? match : `import("${spec}")`));
```
[the remark this answers is not in the document]
Comment (prisis) [reply packages/codegen/src/emit.ts]: Fixed in 32b627dd3. The pattern is gone — the segment is located with `lastIndexOf` in a single pass, and the path-boundary and `@types` checks are ordinary conditions now.

Measured the reported shape before replacing it: 12,000 `/node_modules/` runs took 1.3s through the old pattern and 0.002ms through the scan. Both the boundary cases and a timing guard are pinned by tests in `emit-api.test.ts`.
